src/js/components/DataTable/DataTable.js

```diff
@@ -283,50 +260,98 @@ const DataTable = ({
-          groups={groups}
-          onSelect={onSelect}
-          pad={normalizeProp(pad, 'footer')}
-          pin={pin === true || pin === 'footer'}
-          primaryProperty={primaryProperty}
-          selected={selected}
-          size={size}
+        {groups ? (
+          <GroupedBody
+            background={normalizeProp(background, 'body')}
+            border={normalizeProp(border, 'body')}
+            columns={columns}
+            groupBy={groupBy.property ? groupBy.property : groupBy}
+            groups={groups}
+            groupState={groupState}
+            pad={normalizeProp(pad, 'body')}
+            primaryProperty={primaryProperty}
+            onToggle={onToggleGroup}
+            size={size}
+          />
+        ) : (
+          <Body
+            background={normalizeProp(background, 'body')}
+            border={normalizeProp(border, 'body')}
+            columns={columns}
+            data={!paginate ? adjustedData : currentItems}
+            onMore={onMore}
+            replace={replace}
+            onClickRow={onClickRow}
+            onSelect={
+              onSelect
+                ? nextSelected => {
+                    setSelected(nextSelected);
+                    if (onSelect) onSelect(nextSelected);
+                  }
+                : undefined
+            }
+            pad={normalizeProp(pad, 'body')}
+            pinnedBackground={normalizeProp(background, 'pinned')}
+            placeholder={placeholder}
+            primaryProperty={primaryProperty}
+            rowProps={rowProps}
+            selected={selected}
+            size={size}
+            step={step}
+          />
+        )}
+        {showFooter && (
+          <Footer
+            ref={footerRef}
+            background={normalizeProp(background, 'footer')}
+            border={normalizeProp(border, 'footer')}
+            columns={columns}
+            fill={fill}
+            footerValues={footerValues}
+            groups={groups}
+            onSelect={onSelect}
+            pad={normalizeProp(pad, 'footer')}
+            pin={pin === true || pin === 'footer'}
+            primaryProperty={primaryProperty}
+            selected={selected}
+            size={size}
+          />
+        )}
+        {placeholder && (
+          <StyledPlaceholder top={headerHeight} bottom={footerHeight}>
+            {typeof placeholder === 'string' ? (
+              <Box
+                background={{ color: 'background-front', opacity: 'strong' }}
+                align="center"
+                justify="center"
+                fill="vertical"
+              >
+                <Text>{placeholder}</Text>
+              </Box>
+            ) : (
+              placeholder
+            )}
+          </StyledPlaceholder>
+        )}
+      </StyledDataTable>
+      {paginate && currentItems && (
+        <Pagination
+          direction="row"
+          numItems={data.length}
+          justify={paginate.justify || 'end'}
```

Comment:
I'm wondering whether `justify` should be the responsibility of Pagination or of the caller. If it's up to the caller, then the caller can use a Footer with the custom controls on one side and Pagination on the other.

Reply:
Updated in recent changes, the caller decides this